### ISSUE TITLE
Python3 support

### ISSUE THREAD

**Issue by kpj** [simonvh/fluff#69]
Hello,

Is Python3.x support planned for the future (possibly using `__future__` and [`six`](https://pypi.python.org/pypi/six) to ease the transition)?
From quickly skimming through the code, I'd assume that mainly print-statement, possibly some divisions and the usage of `colorbrewer` would require adjustments.

What do you think?

**Comment by simonvh** [comment]
Actually this is indeed a good idea, given the end of life of Python 2. @BioNinja actually has made a personal Python 3 branch for fluff, and I plan on looking at that when I find some time.

**Comment by kpj** [comment]
Nice, that's great!
Are you interested in help/pull requests?

**Comment by BioNinja** [comment]
my py3 branch  used palettable instead of colorbrewer and only works for py3. 

I think the main transition code would be focus on this issues:

* metaclass: use six to make py2 and py3 work
* temfile: need extra code lines, which made the source code ugly.
* cmp(): py3 did not have this function anymore. 

Numpy is going to stop py2 support, I think we should do the same.

Should I make a pull requests to the main repo? @simonvh @kpj

**Comment by simonvh** [comment]
Thanks for the input @BioNinja. I am totally fine with dropping Python 2 support. Maintaining scientific software is hard enough as it is, I think supporting py3 only is fine. The stable Python 2 version of fluff will be still usable anyway. 

@BioNinja: I've created a py3 branch, can you [update your pull request](https://github.com/blog/2224-change-the-base-branch-of-a-pull-request) to this branch? 
@kpj: Yes, help would be greatly appreciated, as I don't have too much time available at the moment to devote to this. 

Basically, if the tests pass on Travis that is already a sign that most functionality works. However, the test coverage is low at the moment, so quite some functionality is not tested. This is an area that would need work. So, testing the py3 branch would be great.

**Comment by BioNinja** [comment]
@simonvh , done. Hope py3 branch works the same as master.

**Comment by simonvh** [comment]
Thanks! It is (currently) exactly the same as master. This means we can test this branch while keeping the master branch stable.

**Comment by kpj** [comment]
I am having troubles installing the Python3-version of fluff due to the PyCluster dependency.
```bash
$ pip install --user pycluster
Collecting pycluster
  Could not find a version that satisfies the requirement pycluster (from versions: )
No matching distribution found for pycluster
```
How are you dealing with this?

In any case, what do you think about using sklearn/scipy clustering methods instead?

**Comment by BioNinja** [comment]
use conda to install pycluster：

    conda install pycluster -c bio conda

**Comment by zfrenchee** [comment]
Is it possible to install the python3 version with pip? I'm not seeing a py3 wheel on [pypi](https://pypi.python.org/pypi/biofluff/2.1.4).

**Comment by simonvh** [comment]
@zfrenchee Nope, the python 3 version is not released on pypi or conda. I still have to release it, but it has been a very busy period for me and I did not find the time to do so. However, you should be able to install the Python 3 version with pip as follows: 

    pip install git+https://github.com/simonvh/fluff.git@py3

@kpj Yes, I should ditch the pycluster dependency, see #38

**Comment by simonvh** [comment]
Python 3 is now supported via conda, and Pycluster is replaces with KMeans from sklearn. Closing due to inactivity, feel free to re-open if necessary.
